src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -109,7 +109,7 @@ class Docker implements Serializable {
         }
 
         public String imageName() {
-            return toQualifiedImageName(id)
+            return toQualifiedImageName(parsedId.userAndRepo + ":" + parsedId.tag)
```

Comment:
If the raw image id already has a registry in this, the toQualifiedImageName() code would prepend another one, as it expects to *always* get a userRepoTag-like id to operate on.  Now that the parsedId separates the registry from the userAndRepo part, we can pass the correct information along to DockerRegistryEndpoint.imageName()